### PR TITLE
Refactor event log repos to remove inheritence

### PR DESCRIPTION
**Pull request by skiddykong**

### CODE REVIEW

File: example-context/example-service/example-it/src/test/java/uk/gov/justice/services/eventsourcing/repository/jdbc/event/EventRepositoryFactory.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+public class EventRepositoryFactory {
+
+    public static EventJdbcRepository getEventJdbcRepository(final DataSource dataSource) throws NamingException {
+        final EventJdbcRepository eventJdbcRepository = new EventJdbcRepository();
+        eventJdbcRepository.dataSource = dataSource;
+        eventJdbcRepository.jdbcDataSourceProvider = new JdbcDataSourceProvider();
+        eventJdbcRepository.logger = getLogger(EventJdbcRepository.class);
+        eventJdbcRepository.eventJdbcRepositoryHelper = new JdbcRepositoryHelper<>();
+        eventJdbcRepository.uuidJdbcRepositoryHelper = new JdbcRepositoryHelper<>();
+        eventJdbcRepository.eventInsertionStrategy = new AnsiSQLEventLogInsertionStrategy();
```
Comment (matt-rich) [inline example-context/example-service/example-it/src/test/java/uk/gov/justice/services/eventsourcing/repository/jdbc/event/EventRepositoryFactory.java]: Does this need to support a postgres insertion strategy ?

File: persistence/persistence-jdbc/src/main/java/uk/gov/justice/services/jdbc/persistence/JdbcDataSourceProvider.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import javax.sql.DataSource;
+
+public class JdbcDataSourceProvider {
+
+    private static final String JNDI_DS_EVENT_STORE_PATTERN = "java:/app/%s/DS.eventstore";
+
+    @Resource(lookup = "java:app/AppName")
+    String warFileName;
```
Comment (matt-rich) [inline persistence/persistence-jdbc/src/main/java/uk/gov/justice/services/jdbc/persistence/JdbcDataSourceProvider.java]: Can this be private?

File: persistence/persistence-jdbc/src/main/java/uk/gov/justice/services/jdbc/persistence/JdbcDataSourceProvider.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                datasource = (DataSource) getInitialContext().lookup(jndiName());
+            } catch (final NamingException e) {
+                throw new JdbcRepositoryException(e);
+            }
+        }
+        return datasource;
+    }
+
+    public String jndiName() {
+        return String.format(JNDI_DS_EVENT_STORE_PATTERN, warFileName);
```
Comment (matt-rich) [inline persistence/persistence-jdbc/src/main/java/uk/gov/justice/services/jdbc/persistence/JdbcDataSourceProvider.java]: static import

File: persistence/persistence-jdbc/src/main/java/uk/gov/justice/services/jdbc/persistence/JdbcRepositoryHelper.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import javax.sql.DataSource;
+
+public class JdbcRepositoryHelper<T> {
```
Comment (matt-rich) [inline persistence/persistence-jdbc/src/main/java/uk/gov/justice/services/jdbc/persistence/JdbcRepositoryHelper.java]: Maybe have a javadoc comment explaining what this is and the Type.

File: persistence/persistence-jdbc/src/main/java/uk/gov/justice/services/jdbc/persistence/JdbcRepositoryHelper.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+    /**
+     * @param psWrapper               prepared statement wrapper
+     * @param resultSet               jdbc resultSet
+     * @param resultSetToEntityMapper - function mapping resultSet to entity of type U
+     * @param <U>                     - generic type of entity
+     * @return stream of entities
+     */
+    public <U> Stream<U> spliteratorStreamOf(final PreparedStatementWrapper psWrapper, final ResultSet resultSet, final Function<ResultSet, U> resultSetToEntityMapper) {
+        return StreamSupport.stream(new Spliterators.AbstractSpliterator<U>(
```
Comment (matt-rich) [inline persistence/persistence-jdbc/src/main/java/uk/gov/justice/services/jdbc/persistence/JdbcRepositoryHelper.java]: Can this be an abstract class? Is it slightly cleaner?

File: .gitignore [hunk 6]
```diff
@@ -18,6 +18,8 @@ pom.xml~
 .gitignore~
 activemq-data/
 .sonar
+cxf-rs.properties
+event-sourcing/event-source-api-integration-test/
```
[the remark this answers is not in the document]
Comment (skiddykong) [reply .gitignore]: lol

File: aggregate-snapshot/aggregate-snapshot-service/pom.xml [hunk 7]
```diff
@@ -100,4 +100,31 @@
             <scope>test</scope>
         </dependency>
     </dependencies>
+
+    <build>
```
[the remark this answers is not in the document]
Comment (skiddykong) [reply aggregate-snapshot/aggregate-snapshot-service/pom.xml]: Not an obvious one

File: example-context/example-service/example-it/src/test/java/uk/gov/justice/services/eventsourcing/repository/jdbc/event/EventRepositoryFactory.java (hunk 1 above)
Comment (skiddykong) [reply example-context/example-service/example-it/src/test/java/uk/gov/justice/services/eventsourcing/repository/jdbc/event/EventRepositoryFactory.java]: As discussed, we will raise separate ticket to sort this out

File: persistence/persistence-jdbc/src/main/java/uk/gov/justice/services/jdbc/persistence/JdbcRepositoryHelper.java (hunk 4 above)
Comment (skiddykong) [reply persistence/persistence-jdbc/src/main/java/uk/gov/justice/services/jdbc/persistence/JdbcRepositoryHelper.java]: done

File: persistence/persistence-jdbc/src/main/java/uk/gov/justice/services/jdbc/persistence/JdbcRepositoryHelper.java (hunk 5 above)
Comment (skiddykong) [reply persistence/persistence-jdbc/src/main/java/uk/gov/justice/services/jdbc/persistence/JdbcRepositoryHelper.java]: Separate ticket around refactoring that

File: components/event/event-listener/pom.xml [hunk 8]
```diff
@@ -100,4 +100,31 @@
         </dependency>
     </dependencies>
 
+    <build>
```
[the remark this answers is not in the document]
Comment (skiddykong) [reply components/event/event-listener/pom.xml]: Separate ticket

File: event-sourcing/event-repository/event-repository-jdbc/src/test/java/uk/gov/justice/services/eventsourcing/repository/jdbc/event/EventJdbcRepositoryIT.java [hunk 9]
```diff
@@ -39,17 +44,37 @@
...
+            jdbcRepository.eventInsertionStrategy = new AnsiSQLEventLogInsertionStrategy();
+
+            final Poller poller = new Poller();
+
+            poller.pollUntilFound(() -> {
+                try {
+                    jdbcRepository.dataSource.getConnection().prepareStatement("SELECT COUNT (*) FROM event_log;").execute();
+                    return Optional.of("Success");
+                } catch (SQLException e) {
+                    e.printStackTrace();
```
[the remark this answers is not in the document]
Comment (skiddykong) [reply event-sourcing/event-repository/event-repository-jdbc/src/test/java/uk/gov/justice/services/eventsourcing/repository/jdbc/event/EventJdbcRepositoryIT.java]: Done

**Pull request merged**